tests/test_inner.py

```diff
@@ -891,21 +943,16 @@ def test_cancel_build(request, fail_at):
     """
     Verifies that exit plugins are executed when the build is canceled
     """
-
-    # Make the phase we're testing send us SIGTERM
-    phase_signal = defaultdict(lambda: None)
-    phase_signal[fail_at] = signal.SIGTERM
-
+    flexmock(DockerfileParser, content='df_content')
     this_file = inspect.getfile(PreRaises)
     mock_docker()
-    build_timeout = 10 if fail_at == 'build' else 0
-    sigterm_timeout = 2
-    fake_builder = MockInsideBuilder(timeout=build_timeout)
+    fake_builder = MockInsideBuilder()
     flexmock(InsideBuilder).new_instances(fake_builder)
-    watch_pre = WatcherWithSignal(phase_signal['pre'])
-    watch_prepub = WatcherWithSignal(phase_signal['prepub'])
-    watch_post = WatcherWithSignal(phase_signal['post'])
-    watch_exit = WatcherWithSignal(phase_signal['exit'])
+    watch_pre = WatcherWithSignal(signal.SIGTERM)
```

Comment:
yeah it did the same but this way is more clear, as there is only 1 signal anyway used, no need to complicate things